project-koku/koku
# [COST-272] Filter on tag key/value

Previously there was not a filter available for tag keys or values and now there is. At the tags endpoint for each provider, you can filter on a key (/aws/?filter[key]=version) and after going to a specific key you can filter the values (/aws/version/?filter[value]=prod). It supports partial matching of keys and values.

Added many to many field in the tags tables in order to make them filterable.


Testing:
1. load some data in for each provider `make create-test-customer` and `make load-test-customer-data` 
2. Go to the tags endpoint for a provider such as aws, `tags/aws/` and see all the tags and values for that provider
3. Pick one of the tags and try to filter on that tag, using the aws example: `tags/aws/?filter[key]=version` and you should see the values: prod, master, gamma and beta.
4. Go to the endpoint for that specific tag and see the same values as above, `tags/aws/version/`
5. Filter for a specific value or a partial match, `tags/aws/version/?filter[value]=prod` and see the tag or matching tags if you do a partial match
6. repeat for the other providers

Smokes results: 
```
============================================================================ short test summary info =============================================================================
FAILED iqe_cost_management/tests/rest_api/v1/test_cost_model.py::test_api_cost_model_markup_calculation_aws - assert 0.0 != 0.0
==================================================== 1 failed, 3946 passed, 81 skipped, 1729 deselected in 1341.54s (0:22:21) ====================================================
```
This test is unrelated and fails on master.
